consensus/core/src/network/anemo_network.rs

```diff
@@ -0,0 +1,335 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::{collections::BTreeMap, net::Ipv4Addr, sync::Arc, thread::sleep, time::Duration};
+
+use anemo::Response;
+use anemo_tower::auth::{AllowedPeers, RequireAuthorizationLayer};
+use arc_swap::ArcSwapOption;
+use async_trait::async_trait;
+use bytes::Bytes;
+use consensus_config::{AuthorityIndex, NetworkKeyPair};
+use fastcrypto::traits::KeyPair as _;
+use mysten_network::multiaddr::Protocol;
+use tracing::error;
+
+use self::anemo_gen::consensus_rpc_server::ConsensusRpc;
+
+use super::{
+    FetchBlocksRequest, FetchBlocksResponse, NetworkClient, NetworkManager, NetworkService,
+    SendBlockRequest, SendBlockResponse,
+};
+use crate::{
+    block::BlockRef,
+    context::Context,
+    error::{ConsensusError, ConsensusResult},
+};
+
+// Anemo generated traits for RPCs.
+mod anemo_gen {
+    include!(concat!(env!("OUT_DIR"), "/consensus.ConsensusRpc.rs"));
+}
+use anemo_gen::{
+    consensus_rpc_client::ConsensusRpcClient, consensus_rpc_server::ConsensusRpcServer,
+};
+
+pub(crate) struct AnemoClient {
+    inner: Arc<AnemoClientInner>,
+}
+
+struct AnemoClientInner {
+    context: Arc<Context>,
+    network: ArcSwapOption<anemo::Network>,
+}
```

Comment:
Don't have an issue with it per se, but just curious why we need the `AnemoClientInner` and not just have the fields defined directly in `AnemoClient`.  I can see cases where the inner struct is guarded around a lock , but here I don't see too much value (given that `AnemoClient` will already be shared via an Arc it self)

Reply:
Either way is fine. I thought there could be more fields, so it would be better to hold them in a struct and have an Arc<> around. Updated.